lib/reverse_sentence.rb

```diff
@@ -1,6 +1,38 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2), where n is the length of my_sentence
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+  return "" if my_sentence == ""
+  my_sentence = reverse_words(my_sentence)
+  my_sentence = string_reverse(my_sentence, 0, my_sentence.length - 1)
+  return my_sentence
+end
+
+def reverse_words(my_words)
+  i = 0
+  while i < my_words.length
+    if my_words[i] != " "
```

Comment:
Consider using a method like IsWhitespace() instead of comparing to " ". Requirements may change so that other characters like TAB are also counted as whitespace.